### PR TITLE
Terraform refactory: generic node definition and node labelling/taint 

### PR DESCRIPTION
**Pull request by andersla**
Changes:

- Node,edge and master scripts are merged into single node script called multiple times from main.tf
- Changed gce nodes to always use extra disk (size 1gb if 0 or no size specified)
(this is because getting terraform and gce api to do conditional adding extra disk did not work)
- Optional posibility to specify excisting subnet in aws.
- Optional to use master as edge.
- Moved ansible inventory generation from node script to main script.
- Added terraform cloudflare module.
- Added triggers to ansible inventory generator.
- Deleted old edge and master modules (now replaced with general node module).

### CODE REVIEW

File: aws/main.tf [hunk 1]
```diff
@@ -46,48 +58,155 @@ module "vpc" {
...
+  bootstrap_file = "bootstrap/node.sh"
+  master_ip = "${element(module.master.local_ip_v4, 0)}"
+}
+
+#
+# The code below should be identical for all cloud providers
+#
+
+# set cloudflare record (optional): only if var.cloudflare_domain != "" 
+# currentlty this is always including master as edge - but instead should probably be called twice
```
Comment (mcapuccini) [inline aws/main.tf]: it's optional now :)

**Comment by mcapuccini** [comment]
@andersla I am going to trim it down to:

1. Generic node definition for less copy/paste
2. Node labeling and taint

Please send more PRs later on for the other functionalities.

**Comment by mcapuccini** [comment]
@andersla I am now done with the first round of review/revision. Before proceeding I will fix the packer CI, so I can trigger tests. Also I was thinking if it would make sense to have two separated node modules in GCE: `node` and `node-extra-disk`. I think that little copy paste is tolerable until HashiCorp improves the GCE node resource. If you think it make sense could you do this mod?

**Comment by andersla** [comment]
Looks great! I was considering that as well (with two node modules for gce), I will do it.

**Comment by mcapuccini** [comment]
I moved this to https://github.com/kubenow/KubeNow/pull/160

**Pull request closed without merging**